src/islpr.cc

```diff
@@ -0,0 +1,401 @@
+
+#include <iostream>
+#include <iegenlib.h>
+#include <cstring>
+//#include "/home/king/tools/IEGenLib/src/set_relation/set_relation.h"
+
+using namespace std;
+
+string ISLsubtract (string relstr, string knownstr)
+{
+  // Get an isl context
+  isl_ctx *ctx;
+  ctx = isl_ctx_alloc();
+  
+  // Get an isl printer and associate to an isl context
+  isl_printer * ip = NULL;
+  ip = isl_printer_to_str(ctx);
+  
+  // load Relation r into ISL map
+  isl_map* imapR = NULL;
+  isl_map* imapK = NULL;
+  imapR = isl_map_read_from_str(ctx, relstr.c_str());
+  imapK = isl_map_read_from_str(ctx, knownstr.c_str());
+
+  imapR = isl_map_union( imapR, imapK);
+
+  // get string back from ISL map
+  char * cstr;
+  isl_printer_set_output_format(ip , ISL_FORMAT_ISL);
+  isl_printer_print_map(ip ,imapR);
+  cstr=isl_printer_get_str(ip);
+  string stringFromISL = cstr;
+/*  
+  // clean-up
+  isl_printer_flush(ip);
+  isl_printer_free(ip);
+  free(cstr);
+  isl_map_free(imapR);
+  imapR= NULL;
+  isl_map_free(imapK);
+  imapK= NULL;
+  isl_ctx_free(ctx); 
+*/
+  return stringFromISL;
+
+}
+
+string getRelationStringFromISL(string relstr)
```

Comment:
You don't need another copy of this function.  We already have it in set_relation.cc right?